core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProviderSpec.scala

```diff
@@ -317,6 +319,55 @@ class LocalEntityProviderSpec
         )
     }
 
+    "accept multiple update operations for the same entity in batchUpsert" in withLocalEntityProviderTestDatabase {
+      dataSource =>
+        val workspaceContext = runAndWait(
+          dataSource.dataAccess.workspaceQuery.findById(localEntityProviderTestData.workspace.workspaceId)
+        ).get
+        val localEntityProvider = new LocalEntityProvider(EntityRequestArguments(workspaceContext, testContext),
+                                                          slickDataSource,
+                                                          cacheEnabled = true,
+                                                          workbenchMetricBaseName
+        )
+
+        val multiUpsert = Seq(
+          EntityUpdateDefinition("myname",
+                                 "mytype",
+                                 Seq(AddUpdateAttribute(AttributeName.withDefaultNS("one"), AttributeString("111")))
+          ),
+          EntityUpdateDefinition("myname",
+                                 "mytype",
+                                 Seq(AddUpdateAttribute(AttributeName.withDefaultNS("two"), AttributeString("222")))
+          )
+        )
+        val writes = localEntityProvider.batchUpsertEntities(multiUpsert).futureValue
+
+        writes.size shouldBe 2
+
+        val entityQuery = EntityQuery(1, 100, "name", SortDirections.Ascending, None)
+        val parentContext = RawlsRequestContext(userInfo)
+        val actual = localEntityProvider.queryEntities("mytype", entityQuery, parentContext).futureValue
+
+        actual.resultMetadata.unfilteredCount shouldBe 1
+        actual.results.size shouldBe 1
+        actual.results.head shouldBe Entity("myname",
+                                            "mytype",
+                                            Map(AttributeName.withDefaultNS("one") -> AttributeString("111"),
+                                                AttributeName.withDefaultNS("two") -> AttributeString("222")
+                                            )
+        )
+
+        val withAllAttrs = runAndWait(
+          dataSource.dataAccess.entityQueryWithInlineAttributes
+            .findEntityByName(localEntityProviderTestData.workspace.workspaceIdAsUUID, "mytype", "myname")
```

Comment:
Just trying to understand - what's the difference between these two queries - `localEntityProvider.queryEntities` and `dataSource.dataAccess.entityQueryWithInlineAttributes`?

Reply:
The major difference between the two is that the `*WithInlineAttributes` query returns the `ENTITY.all_attribute_values` column, while `queryEntities` does not. Since `all_attribute_values` can be large - kilobytes or even megabytes in a pathological case - we avoid returning it from the db whenever possible. In this particular test, I explicitly want to check that its value is being calculated correctly.